drivers/platform/stm32/stm32_i3c.c

```diff
@@ -637,6 +673,246 @@ int stm32_i3c_get_ccc_info(struct no_os_i3c_bus_desc* desc,
+	tx_count = 0;
+	rx_count = 0;
+
+	/* DMA transfer is not supported if DMA descriptor iunavailable */
+	if (!xdesc->i3c_dma_desc)
+		return -ENOTSUP;
+	/* A transfer is ongoing if private data exists */
+	if (xdesc->i3c_dma_desc->priv)
+		return -EBUSY;
+
+	/* check on the Mode */
+	if (hi3c->Mode != HAL_I3C_MODE_CONTROLLER) {
+		hi3c->ErrorCode = HAL_I3C_ERROR_NOT_ALLOWED;
+		return -ENOTSUP;
+	}
+	/* Check on hdmacr handle */
+	else if (xdesc->i3c_dma_desc->crdma_ch == NULL) {
+		hi3c->ErrorCode = HAL_I3C_ERROR_DMA_PARAM;
+		return -EINVAL;
+	}
+	/* check on the State */
+	else if ((hi3c->State != HAL_I3C_STATE_READY)
+		 && (hi3c->State != HAL_I3C_STATE_LISTEN)) {
+		return -EBUSY;
+	}
+
+	/* Verify and count the messages */
+	for (i = 0; i < len; i++) {
+		if (((msgs[i].tx_buff == NULL) && (msgs[i].tx_size != 0U)) ||
+		    ((msgs[i].rx_buff == NULL) && (msgs[i].rx_size != 0U))) {
+			return -EINVAL;
+		}
+		if (msgs[i].tx_buff)
+			tx_count++;
+		if (msgs[i].rx_buff)
+			rx_count++;
+	}
+
+	if (tx_count && !xdesc->i3c_dma_desc->txdma_ch)
+		return -EINVAL;
+	if (rx_count && !xdesc->i3c_dma_desc->rxdma_ch)
+		return -EINVAL;
+	dma_ctx = (struct stm32_i3c_dma_ctx *) no_os_calloc(1,
+			sizeof(struct stm32_i3c_dma_ctx));
+	if (!dma_ctx)
+		return -ENOMEM;
+
+	/*********************** Fill the DMA Context **************************/
+	xdesc->i3c_dma_desc->priv = (void *) dma_ctx;
+	ret = stm32_i3c_populate_dma_context(desc, msgs, len, callback, ctx);
+	if (ret)
+		goto free_dma_ctx;
+	/********************** I3C Handle Configuration ************************/
+
+	/* Set handle transfer parameters */
+	hi3c->ErrorCode     = HAL_I3C_ERROR_NONE;
+	hi3c->XferISR       = stm32_i3c_xfer_complete_cb;
+	hi3c->ControlXferCount = len;
+
+	if ((tx_count == 0) && (rx_count == 0))
+		hi3c->State         = HAL_I3C_STATE_BUSY;
+	else if (rx_count == 0)
+		hi3c->State         = HAL_I3C_STATE_BUSY_TX;
+	else if (tx_count == 0)
+		hi3c->State         = HAL_I3C_STATE_BUSY_RX;
+	else
+		hi3c->State         = HAL_I3C_STATE_BUSY_TX_RX;
+
+	/* Enable Tx process (frame complete and error) interrupts */
+	hi3c->Instance->IER |= (HAL_I3C_IT_FCIE | HAL_I3C_IT_ERRIE);
+
+	/* Enable control DMA Request */
+	LL_I3C_EnableDMAReq_Control(hi3c->Instance);
+
+	/* Enable Rx data DMA Request */
+	if (rx_count != 0U) {
+		LL_I3C_EnableDMAReq_RX(hi3c->Instance);
+	}
+
+	/* Enable Tx data DMA Request */
+	if (tx_count != 0U) {
+		LL_I3C_EnableDMAReq_TX(hi3c->Instance);
+	}
+
+	/*********************** I3C DMA Configuration ************************/
+	ret = stm32_i3c_dma_config(xdesc, dma_ctx, tx_count, rx_count);
+	if (ret)
+		goto abort_transfer;
+	/********************** Priority Configuration ************************/
+	/*
+	 * Note: The priority of the I3C interrupt has to lower than DMA interrupt.
+	 * The I3C interrupt is made to wait until the DMA interrupt is received
+	 * and then continues to write the next control word once the DMA transfer is
+	 * configured.
+	 */
```

Comment:
The priority of the I3C interrupt has to be lower than DMA interrupt.
or
 The priority of the I3C interrupt is lower than DMA interrupt.
?
```suggestion
        /*
	 * The priority of the I3C interrupt has to be lower than the DMA interrupt.
	 * The I3C interrupt waits until the DMA interrupt is received
	 * and then continues to write the next control word once the DMA transfer is
	 * configured.
	 */
```